### ISSUE TITLE
Sharing compiled graphs between different weights without incurring loading cost

### ISSUE THREAD

**Issue by isidentical** [siliconflow/onediff#581]
Same compiled graphs can be individually loaded into different pipelines, but loading them has a cost attached (around 15-20 second for the unet/fast_unet/vae component graph to be read from the disk and applied). Is it possible to somehow mitigate this for the same graphs used in different pipeline/weights?

```py
import torch
import time
import os
from pathlib import Path

os.environ["ONEFLOW_MLIR_ENABLE_INFERENCE_OPTIMIZATION"] = "1"

from diffusers_extensions.deep_cache import StableDiffusionXLPipeline
from onediff.infer_compiler import oneflow_compile
from onediff.schedulers import EulerDiscreteScheduler

GRAPHS_DIR = Path("/data/oneflow-cache/graphs/global-sdxl/1/onediff-ce-0.12.1/")

def load_compiled_graphs(pipe):
    if pipe.vae.dtype == torch.float16 and pipe.vae.config.force_upcast:
        pipe.upcast_vae()

    for component_name in ["unet", "fast_unet", "vae"]:
        component = getattr(pipe, component_name)
        component = oneflow_compile(component)
        target_file = GRAPHS_DIR / f"{component_name}-compiled.of"
        print(f"Loading compiled graph for {component_name}")
        component.load_graph(target_file)
        setattr(pipe, component_name, component)

scheduler = EulerDiscreteScheduler.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    subfolder="scheduler",
)

# Initialize the first model
pipe_1 = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.float16,
    scheduler=scheduler,
    variant="fp16",
)
pipe_1.to("cuda")
t0 = time.perf_counter()
load_compiled_graphs(pipe_1)
t1 = time.perf_counter()
print("Loading compiled graphs took", t1 - t0, "seconds")

# Initialize the second model
pipe_2 = StableDiffusionXLPipeline.from_pretrained(
    "dataautogpt3/OpenDalleV1.1",
    torch_dtype=torch.float16,
    scheduler=scheduler,
)
pipe_2.to("cuda")
t0 = time.perf_counter()
load_compiled_graphs(pipe_2)
t1 = time.perf_counter()

# How to make this take less time? Or re-use the same graphs from pipe1 which are
# already loaded?
print("Loading compiled graphs took", t1 - t0, "seconds")
```

**Comment by strint** [comment]
@isidentical 

This example shows how to share compiled components between pipelines: https://github.com/siliconflow/onediff/pull/644

Replacing warmup by run to warmup by loading should be the feature you want.

**Comment by strint** [comment]
Another way to share the same graph for different weights is by switching weights. Switching weight should be the same as [Switching LoRAs](https://github.com/siliconflow/onediff/tree/main/onediff_diffusers_extensions#fast-lora-loading-and-switching) because the graph shares the same parameters with the nn.Module in the pipeline. 

We will try to create a example for sharing the same graph for different weights later.

**Comment by isidentical** [comment]
@strint, this seems to only allow sharing non-finetuned parts of the compiled graphs, but the biggest time wasting of unet is till there. Is there a way we can share the underlying compiled graph in unet while changing the weights?

**Comment by strint** [comment]
@isidentical  Have you tried to update the weight of the same pipe instance without onediff?

**Comment by strint** [comment]
> @strint, this seems to only allow sharing non-finetuned parts of the compiled graphs, but the biggest time wasting of unet is till there. Is there a way we can share the underlying compiled graph in unet while changing the weights?

@isidentical Here is a draft demo for this: https://github.com/siliconflow/onediff/pull/647

**Comment by strint** [comment]
closed with: https://github.com/siliconflow/onediff/pull/647
